### ISSUE TITLE
Create ownership contract

### ISSUE THREAD

**Issue by shasikaud** [Reppo-Labs/Reppo-Protocol#38]
- [x] setup hardhat
- [x] create ownership contract
- [x] create sample multisig contract
- [x] add test suites
- [ ] deploy to base testnet
- [x] add events

**Comment by trvrggrep** [comment]
Can the multisig address be any multisig framework, like gnosis safe? Or, it has to be custom with the updateRecord function?

**Comment by trvrggrep** [comment]
Does it make sense to create a separate repo for these contracts? Would it make more sense to use reppo-protocol repo instead?

**Comment by trvrggrep** [comment]
Update the following field names:

modelId ->> podid
description ->> podDescription
multiSigContract ->>updateAdmin
percentages ->> ownershipPercentages

**Comment by shasikaud** [comment]
> Can the multisig address be any multisig framework, like gnosis safe? Or, it has to be custom with the updateRecord function?

it can be any valid address on the network, no validation is made by our side (ownership contract)

**Comment by trvrggrep** [comment]
```
    it("Throws an error when update is called by non-multisig", async function () {
      const { ownership, owner } = await loadFixture(deployOwnership);
      const { multisig } = await loadFixture(deployMultisig);
      const modelId = "model01";
      const description = "model 01 description";
      const multiSigContract = multisig.target;
      const owners = [owner.address];
      const percentages = [100n];
      await ownership.createRecord(
        modelId,
        description,
        multiSigContract,
        owners,
        percentages
      );
      const newDescription: string = "new description";
      await expect(ownership.connect(owner).updateRecord(
        modelId,
        newDescription,
        owners,
        percentages,
      )).to.be.revertedWith("Only multisig contract can update record");
    });
```

Just want to get clarity for this ^ test case. It should be that only the updateAdmin account can make changes to the record. I think this is what you have implemented, correct?

**Comment by trvrggrep** [comment]
> > Does it make sense to create a separate repo for these contracts? Would it make more sense to use reppo-protocol repo instead?
> 
> ill move the contracts under the parent repo. once permissions are sorted

You don't have correct permissions for the protocol repo? We can chat about this in today's standup

**Comment by shasikaud** [comment]
[Repo](https://github.com/Reppo-Labs/Reppo-Protocol/tree/feature/ownership-contract)
moved the contracts & test suites under `hardhat` folder

**Comment by shasikaud** [comment]
> ```
>     it("Throws an error when update is called by non-multisig", async function () {
>       const { ownership, owner } = await loadFixture(deployOwnership);
>       const { multisig } = await loadFixture(deployMultisig);
>       const modelId = "model01";
>       const description = "model 01 description";
>       const multiSigContract = multisig.target;
>       const owners = [owner.address];
>       const percentages = [100n];
>       await ownership.createRecord(
>         modelId,
>         description,
>         multiSigContract,
>         owners,
>         percentages
>       );
>       const newDescription: string = "new description";
>       await expect(ownership.connect(owner).updateRecord(
>         modelId,
>         newDescription,
>         owners,
>         percentages,
>       )).to.be.revertedWith("Only multisig contract can update record");
>     });
> ```
> 
> Just want to get clarity for this ^ test case. It should be that only the updateAdmin account can make changes to the record. I think this is what you have implemented, correct?

yes correct, only the updateAdmin address (either a wallet or contract) can call the updateRecord function

**Comment by shasikaud** [comment]
1. model attributes renamed
2. added new field names `ipAccountAddress`
3. updated the test suites

[hardhat](https://github.com/Reppo-Labs/Reppo-Protocol/tree/feature/ownership-contract/hardhat)

**Comment by trvrggrep** [comment]
Onwership contract on base 

https://sepolia.basescan.org/address/0xaa62b53c5717ab6bbd496d5f9da89f3de453b0df
